sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -50,161 +50,162 @@
+static void receive_registration_status();
+static void parse_operation_message(
     char* topic,
     int topic_len,
-    MQTTClient_message const* message,
+    const MQTTClient_message* message,
     az_iot_provisioning_client_register_response* response,
     az_iot_provisioning_client_operation_status* operation_status);
-static int send_operation_query_message(
-    az_iot_provisioning_client_register_response const* response);
-
+static void send_operation_query_message(
+    const az_iot_provisioning_client_register_response* response);
+static void disconnect_client_from_provisioning_service();
 static void sleep_for_seconds(uint32_t seconds);
-static void print_az_span(char const* span_description, az_span span);
+static void print_az_span(const char* span_description, az_span span);
 
 int main()
+{
+  create_and_configure_client();
+  LOG_SUCCESS("Client created and configured.\n");
+
+  connect_client_to_provisioning_service();
+  LOG_SUCCESS("Client connected to \"%s\".\n", global_provisioning_endpoint_buffer);
+
+  subscribe_client_to_provisioning_service_topics();
+  LOG_SUCCESS("Client subscribed to provisioning service topics.\n");
+
+  register_client_with_provisioning_service();
+  LOG_SUCCESS("Client registered with provisioning service.\n");
+
+  receive_registration_status();
+  LOG_SUCCESS("Client received registration status.\n")
+
+  disconnect_client_from_provisioning_service();
+  LOG_SUCCESS("Client disconnected from provisioning service.\n");
+
+  return 0;
+}
+
+static void create_and_configure_client()
 {
   int rc;
+  az_span endpoint;
+  az_span id_scope;
+  az_span registration_id;
+
+  if (az_failed(rc = read_environment_variables(&endpoint, &id_scope, &registration_id)))
+  {
+    LOG_ERROR("Failed to read evironment variables: az_result return code 0x%04x .\n", rc);
+    exit(rc);
+  }
 
-  // Read in the necessary environment variables and initialize the az_iot_provisioning_client
-  if (az_failed(rc = read_configuration_and_init_client()))
+  if (az_failed(
+          rc = az_iot_provisioning_client_init(
+              &provisioning_client, endpoint, id_scope, registration_id, NULL)))
   {
-    (void)printf(
-        "Failed to read configuration from environment variables: az_result return code 0x%04x .\n",
-        rc);
-    return rc;
+    LOG_ERROR("Failed to initiate provisioning client: az_result return code 0x%04x .\n", rc);
+    exit(rc);
   }
 
-  // Get the MQTT client id used for the MQTT connection
   if (az_failed(
           rc = az_iot_provisioning_client_get_client_id(
-              &provisioning_client, mqtt_client_id, sizeof(mqtt_client_id), NULL)))
+              &provisioning_client, mqtt_client_id_buffer, sizeof(mqtt_client_id_buffer), NULL)))
   {
-    (void)printf("Failed to get MQTT client id: az_result return code 0x%04x .\n", rc);
-    return rc;
+    LOG_ERROR("Failed to get MQTT client id: az_result return code 0x%04x .\n", rc);
```

Comment:
```suggestion
    LOG_ERROR("Failed to get MQTT client id: az_result return code 0x%04x.\n", rc);
```

Reply:
We had discussed on Monday keep a space or using "" prior to the end-of-sentence period when a url, return code, or anything other than a normal word is at the end of a sentence.  This can definitely be changed back for return codes if we don't like the space here.

Reply:
Removing spaces!